lisa/tools/ntttcp.py

```diff
@@ -44,3 +71,204 @@ def get_throughput(self, stdout: str) -> str:
+        run_as_daemon: bool = False,
+        udp_mode: bool = False,
+    ) -> Process:
+        cmd = ""
+        if server_ip:
+            cmd += f" -r{server_ip} "
+        cmd += (
+            f" -P {ports_count} -t {run_time_seconds} -W {warm_up_time_seconds} "
+            f"-C {cool_down_time_seconds} -b {buffer_size}k "
+            f"--show-nic-packets {nic_name} "
+        )
+        if udp_mode:
+            cmd += " -u "
+        if use_epoll:
+            cmd += " -e "
+        if dev_differentiator:
+            cmd += f" --show-dev-interrupts {dev_differentiator} "
+        if run_as_daemon:
+            cmd += " -D "
+        process = self.node.execute_async(
+            f"ulimit -n 204800 && {self.command} {cmd}", shell=True
+        )
+        return process
+
+    def run_as_server(
+        self,
+        nic_name: str,
+        run_time_seconds: int = 120,
+        ports_count: int = 64,
+        buffer_size: int = 64,
+        cool_down_time_seconds: int = 1,
+        warm_up_time_seconds: int = 1,
+        use_epoll: bool = True,
+        server_ip: str = "",
+        dev_differentiator: str = "Hypervisor callback interrupts",
+        run_as_daemon: bool = False,
+        udp_mode: bool = False,
+    ) -> None:
+        # -rserver_ip: run as a receiver with specified server ip address
+        # -P: Number of ports listening on receiver side [default: 16] [max: 512]
+        # -t: Time of test duration in seconds [default: 60]
+        # -e: [receiver only] use epoll() instead of select()
+        # -u: UDP mode     [default: TCP]
+        # -W: Warm-up time in seconds          [default: 0]
+        # -C: Cool-down time in seconds        [default: 0]
+        # -b: <buffer size in n[KMG] Bytes>    [default: 65536 (receiver); 131072
+        # (sender)]
+        # --show-nic-packets <network interface name>: Show number of packets
+        # transferred (tx and rx) through this network interface
+        # --show-dev-interrupts <device differentiator>: Show number of interrupts for
+        # the devices specified by the differentiator
+        # Examples for differentiator: Hyper-V PCIe MSI, mlx4, Hypervisor callback
+        # interrupts
+        process = self.run_as_server_async(
+            nic_name,
+            run_time_seconds,
+            ports_count,
+            buffer_size,
+            cool_down_time_seconds,
+            warm_up_time_seconds,
+            use_epoll,
+            server_ip,
+            dev_differentiator,
+            run_as_daemon,
+            udp_mode,
+        )
+        timeout = run_time_seconds + 10
+        process.wait_result(
+            timeout,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="fail to lanuch ntttcp server",
+        )
+
+    def run_as_client(
+        self,
+        nic_name: str,
+        server_ip: str,
+        threads_count: int,
+        run_time_seconds: int = 120,
```

Comment:
Check if this value can be shorter. The default scenario is to find the throughput, 5 or 10 seconds should be enough.

Reply:
updated into 10 seconds